checkout-ui-custom/checkout6-custom.js

```diff
@@ -285,8 +285,9 @@ const MAX_TIME_EXPIRATION = 1000 * 60 * 5 // 5 minutes
       const canEditAddress = 'add-shipping'
       const checkShipping = setInterval(function () {
         if (
-          b2bCheckoutSettings &&
-          b2bCheckoutSettings.permissions.includes(canEditAddress)
+          (b2bCheckoutSettings &&
+            b2bCheckoutSettings.permissions.includes(canEditAddress)) ||
+          !window.vtexjs.checkout.orderForm.loggedIn
         ) {
```

Comment:
Just one small extra thing: in other parts of this file we're checking if `window.vtexjs` and `window.vtexjs.checkout` exist before accessing them. So it might be safer also do it here:

```js
  const isLoggedIn = window.vtexjs &&
   window.vtexjs.checkout &&
   window.vtexjs.checkout.orderForm &&
   window.vtexjs.checkout.orderForm.loggedIn

  if (!isLoggedIn || 
    (b2bCheckoutSettings &&
      b2bCheckoutSettings.permissions.includes(canEditAddress))) {
```